api/handlers_api.go

```diff
@@ -41,67 +41,46 @@ func NewRequest(session *Session, payload ...[]byte) *Request {
 	return &wsReq
 }
 
-func (r *Request) HandleCreateGame() *mc.Message[mc.RespCreateGame] {
-	game := r.Session.GameManager.AddGame()
-	r.Session.GameUuid = game.Uuid
-
-	hostPlayer := game.CreateHostPlayer(r.Session.ID)
-	r.Session.Player = hostPlayer
-
+func (r *Request) HandleCreateGame() mc.Message[mc.RespCreateGame] {
+	var reqCreateGame mc.Message[mc.ReqCreateGame]
 	resp := mc.NewMessage[mc.RespCreateGame](mc.CodeCreateGame)
-	resp.AddPayload(mc.RespCreateGame{GameUuid: game.Uuid, HostUuid: hostPlayer.Uuid})
-	return &resp
-}
-
-// User will choose the configurations of ships on defence grid.
-// Then the grid is sent to backend and adjustment happens accordingly.
-func (r *Request) HandleReadyPlayer() (*mc.Message[mc.NoPayload], *mb.Game) {
-	var readyPlayerReq mc.Message[mc.ReqReadyPlayer]
-	resp := mc.NewMessage[mc.NoPayload](mc.CodeReady)
 
-	if err := json.Unmarshal(r.Payload, &readyPlayerReq); err != nil {
+	if err := json.Unmarshal(r.Payload, &reqCreateGame); err != nil {
 		resp.AddError(err.Error(), cerr.ConstErrInvalidPayload)
-		return &resp, nil
+		return resp
 	}
 
-	game, player, err := r.Session.GameManager.FindGameAndPlayer(readyPlayerReq.Payload.GameUuid, readyPlayerReq.Payload.PlayerUuid)
-	if err != nil {
-		resp.AddError(err.Error(), cerr.ConstErrReady)
-		return &resp, nil
+	gameDifficulty := reqCreateGame.Payload.GameDifficulty
+	if gameDifficulty != mb.GameDifficultyEasy && gameDifficulty != mb.GameDifficultyHard {
```

Comment:
shouldn't this be `gameDifficulty != easy ||  gameDifficulty != normal || gameDifficulty != hard`?

Reply:
on second thought, it would be nicer to write a function on `GameDifficulty` enum to check for validity of it. something like: `gameDifficulty.isValid()`

Reply:
1. I think that `&&` is correct... So if all those conditions are false, it means that the difficulty is none of them. So imagine a scenario when difficulty is normal. we go into the condition and the code sees, oh difficulty != easy, then false.
2. Transferred it into a func `isDifficultyValid`.